pkg/api/routes.go

```diff
@@ -240,7 +240,7 @@ func (rh *RouteHandler) CheckVersionSupport(response http.ResponseWriter, reques
 	response.Header().Set(constants.DistAPIVersion, "registry/2.0")
 	// NOTE: compatibility workaround - return this header in "allowed-read" mode to allow for clients to
 	// work correctly
-	if rh.c.Config.HTTP.Auth != nil {
```

Comment:
this was always evaluated to true(because httpassword.path which is empty string), sending an auth header with the next value:
"basic realm="

Reply:
:(